lib/sequent/support/database.rb

```diff
@@ -0,0 +1,63 @@
+module Sequent
+  module Support
+    # Offers support operations for a postgres database.
+    #
+    # Class methods do establish their own database connections (and therefore
+    # take in a database configuration). Instance methods assume that a database
+    # connection yet is established.
+    class Database
+      attr_reader :db_config
+
+      def self.create!(db_config)
+        ActiveRecord::Base.establish_connection(db_config.merge('database' => 'postgres'))
+        ActiveRecord::Base.connection.create_database(db_config['database'])
+      end
+
+      def self.drop!(db_config)
+        ActiveRecord::Base.establish_connection(db_config.merge('database' => 'postgres'))
+        ActiveRecord::Base.connection.drop_database(db_config['database'])
+      end
+
+      def self.establish_connections(db_config, view_schema_name)
+        Sequent::Support::BaseViewModel.establish_connection(
```

Comment:
What I mean is that currently in a single transaction the events are stored in the event store and the view is updated in the projections. When you use 2 different connectionpools (one for the projections, and one for the event store) this is no longer possible.

So for instance when the event store fails to update, for instance concurrent updates of aggregates, then the view model is not rolled back because it has its own transaction. We experienced this issue with kaiwa (ask @s0meone)